command/loadbot/execution.go

```diff
@@ -235,6 +280,10 @@ func (l *Loadbot) Run() error {
 
 				return
 			}
+			// initialise block numbers
+			l.metrics.GasMetrics.BlockGasMutex.Lock()
+			l.metrics.GasMetrics.Blocks[receipt.BlockNumber] = GasMetrics{}
+			l.metrics.GasMetrics.BlockGasMutex.Unlock()
```

Comment:
This can be a separate method

Reply:
Moved to dedicated method 0d0dc3410f11c71b098c4238344680a86d0d070e